test/functional/p2p_node_network_limited.py

```diff
@@ -46,6 +49,44 @@ def setup_network(self):
         self.add_nodes(self.num_nodes, self.extra_args)
         self.start_nodes()
 
+    def test_avoid_requesting_old_blocks(self):
+        self.log.info("Test node not requesting blocks beyond limited peer threshold")
+        # The request of blocks further than the NETWORK_NODE_LIMITED threshold causes a direct disconnection.
+        pruned_node = self.nodes[0]
+        miner = self.nodes[1]
+        full_node = self.nodes[2]
+
+        # Connect and generate block to ensure IBD=false
+        self.connect_nodes(1, 2)
+        self.connect_nodes(0, 2)
+        self.generate(miner, 1, sync_fun=self.no_op)
+
+        # Verify all peers are at the same height
+        best_block = miner.getbestblockhash()
+        for node in [full_node, pruned_node]:
+            self.wait_until(lambda: node.getbestblockhash() == best_block, timeout=10)
+
+        # Verify that all peers are out of IBD
+        for node in self.nodes:
+            assert not node.getblockchaininfo()['initialblockdownload']
+
+        # Disconnect full_node from the other two peers (the node will remain out of IBD)
+        self.disconnect_nodes(0, 2)
+        self.disconnect_nodes(1, 2)
+        assert_equal(len(full_node.getpeerinfo()), 0)
+
+        # Mine blocks and sync the pruned node. Surpass the NETWORK_NODE_LIMITED threshold.
+        self.generate(miner, MIN_BLOCKS_TO_KEEP + 12, sync_fun=self.no_op)
```

Comment:
Is the intention here to mimic `NODE_NETWORK_LIMITED_MIN_BLOCKS` from the C++ source code, which is `288`? `MIN_BLOCKS_TO_KEEP` is a different constant defined in both the C++ and Python code which happens to be also `288`. Should `NODE_NETWORK_LIMITED_MIN_BLOCKS` be defined in the Python code (equal to 288, but not necessary equal or tied to `MIN_BLOCKS_TO_KEEP`)?

Reply:
Added `NODE_NETWORK_LIMITED_MIN_BLOCKS` on the last update.